ceph/ceph_admin/osd.py

```diff
@@ -106,3 +106,96 @@ def apply(self, config: Dict) -> None:
             sleep(interval)
 
         raise OSDServiceFailure("OSDs are not up and running in hosts")
+
+    def rm_status(self, config: Dict):
+        """
+        Execute the command ceph orch osd rm status.
+
+        Args:
+            config (Dict): OSD Remove status configuration parameters
+
+        Returns:
+          output, error   returned by the command.
+
+        Example::
+
+            config:
+                command: rm status
+                base_cmd_args:
+                    verbose: true
+                args:
+                    format: json-pretty
+        """
+
+        base_cmd = ["ceph", "orch", "osd", "rm", "status"]
+        if config.get("base_cmd_args"):
+            base_cmd.append(config_dict_to_string(config["base_cmd_args"]))
+
+        if config and config.get("args"):
+            args = config.get("args")
+            base_cmd.append(config_dict_to_string(args))
+
+        return self.shell(args=base_cmd)
+
+    def rm(self, config: Dict):
+        """
+        Execute the command ceph orch osd rm <OSD ID> .
+
+        Args:
+            config (Dict): OSD Remove configuration parameters
+
+        Returns:
+          output, error   returned by the command.
+
+        Example::
+
+            config:
+                command: rm
+                base_cmd_args:
+                    verbose: true
+                pos_args:
+                    - 1
+        """
+
+        base_cmd = ["ceph", "orch", "osd"]
+        if config.get("base_cmd_args"):
+            base_cmd.append(config_dict_to_string(config["base_cmd_args"]))
+        base_cmd.append("rm")
+        osd_id = str(config["pos_args"][0])
+        base_cmd.append(osd_id)
+        self.shell(args=base_cmd)
+
+        check_osd_id_dict = {
+            "args": {"format": "json-pretty"},
+        }
+
+        while True:
+            # "ceph orch osd rm status -f json-pretty"
+            # condition
+            # continue loop if OSD_ID present
+            # if not exit the loop
+            out, _ = self.rm_status(check_osd_id_dict)
+
+            try:
+                status = json.loads(out)
+                for osd_id_ in status:
+                    if osd_id_["osd_id"] == osd_id:
```

Comment:
Missed this one, comparing `str` with `int`, so it breaks out at first iteration.

Reply:
This is addressed as per the discussion.